ui/apps/platform/src/Containers/Administration/Events/SearchFilterDomain.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+import { Divider, Select, SelectOption } from '@patternfly/react-core';
+
+import { domains } from 'services/AdministrationEventsService';
+
+const optionAll = 'All';
+
+type SearchFilterDomainProps = {
+    domain: string | undefined;
+    setDomain: (domain: string | undefined) => void;
+};
+
+function SearchFilterDomain({ domain, setDomain }: SearchFilterDomainProps) {
+    const [isOpen, setIsOpen] = useState(false);
+
+    function onToggle(isOpenArg: boolean) {
+        setIsOpen(isOpenArg);
+    }
```

Comment:
I thought one of us created a reusable hook for this functionality, so we didn't have to rewrite it for every little select/expandable/etc. component.

But I can't find one in the codebase. Did i imagine it?

Reply:
Apparently not. There is one example of `onToggle={setIsOpen}` without separate `onToggle` function, which I will investigate as part of the next step.